ipatests/test_integration/test_automember.py

```diff
@@ -0,0 +1,316 @@
+            'ldapsearch', '-x',
+            '-D', 'cn=directory manager',
+            '-w', master.config.dirman_password,
+            '-b', 'cn=Auto Membership Plugin,cn=plugins,cn=config']
+
+        result = self.master.run_command(
+            search_cmd,
+            raiseonerr=False
+        )
+        assert result.returncode == 0
+        assert "automemberprocessmodifyops: off" in result.stdout_text
+
+        '#testcase setup'
+        cmdlist = [['ipa', 'group-add', '--desc="Dallas"', 'DallasUsers'],
+                   ['ipa', 'automember-add', '--type=group', 'DallasUsers'],
+                   ['ipa', 'automember-add-condition',
+                    '--key=destinationindicator', '--type=group',
+                    '--inclusive-regex=Dallas', 'DallasUsers'],
+                   ['ipa', 'user-add', '--first=first', '--last=user',
+                    'user2', '--setattr=destinationindicator=Dallas']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'DallasUsers'],
+        )
+        assert "user2" in result.stdout_text
+
+        '#Modifying the user group so that it becomes part of new group'
+        cmdlist = [['ipa', 'group-add', '--desc="Houston"', 'HoustonUsers'],
+                   ['ipa', 'automember-add', '--type=group', 'HoustonUsers'],
+                   ['ipa', 'automember-add-condition',
+                    '--key=destinationindicator', '--type=group',
+                    '--inclusive-regex=Houston', 'HoustonUsers'],
+                   ['ipa', 'user-mod', 'user2',
+                    '--setattr=destinationindicator=Houston']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
+
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'HoustonUsers'],
+        )
+        assert "user2" not in result.stdout_text
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'DallasUsers'],
+        )
+        assert "user2" in result.stdout_text
+
+        '#Running the automember-build so that user is part of correct group'
+
+        self.master.run_command(['ipa', 'automember-rebuild', '--users=user2'])
+
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'HoustonUsers'],
+        )
+        assert "user2" in result.stdout_text
+        result = self.master.run_command(
+            ['ipa', 'group-show', 'DallasUsers'],
+        )
+        assert "user2" not in result.stdout_text
+
+        '#testcase cleanup'
+        cmdlist = [['ipa', 'user-del', 'user2'],
+                   ['ipa', 'automember-del', '--type=group', 'DallasUsers'],
+                   ['ipa', 'automember-del', '--type=group', 'HoustonUsers'],
+                   ['ipa', 'group-del', 'DallasUsers'],
+                   ['ipa', 'group-del', 'HoustonUsers']]
+
+        for cmd in cmdlist:
+            self.master.run_command(cmd,)
```

Comment:
For cleanup commands I would add raiseonerr=False.
The test can also be written like this:
try:
[do calls]
finally:
[do cleanup]